### PR TITLE
[improvement](mtmv) Add id to statistics map in statement context for cost estimation later

### PR DESCRIPTION
**Pull request by seawinde**
## Proposed changes

Add id to statistics map in statement context for cost estimation later
this helps to improve the probability to use materialized view when query a single table with aggregate and many filter

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by seawinde** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/StatementContext.java [hunk 1]
```diff
@@ -149,6 +152,11 @@ public class StatementContext implements Closeable {
     // Record table id mapping, the key is the hash code of union catalogId, databaseId, tableId
     // the value is the auto-increment id in the cascades context
     private final Map<TableIdentifier, TableId> tableIdMapping = new LinkedHashMap<>();
+    // Record the materialization statistics by id which is used for cost estimation.
+    // Maybe return null, which means the id according statistics should calc normally rather than getting
+    // form this map
+    // id maybe relation id or cteId or other type of id
+    private final Map<Pair<Id, Class<? extends Id>>, Statistics> idToStatisticsMap = new LinkedHashMap<>();
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/StatementContext.java:159]: just support relation id now

**Comment by seawinde** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/StatementContext.java [hunk 2]
```diff
@@ -149,6 +152,10 @@ public class StatementContext implements Closeable {
     // Record table id mapping, the key is the hash code of union catalogId, databaseId, tableId
     // the value is the auto-increment id in the cascades context
     private final Map<TableIdentifier, TableId> tableIdMapping = new LinkedHashMap<>();
+    // Record the materialization statistics by id which is used for cost estimation.
+    // Maybe return null, which means the id according statistics should calc normally rather than getting
+    // form this map
+    private final Map<RelationId, Statistics> relationIdToStatisticsMap = new LinkedHashMap<>();
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/StatementContext.java:158]: how to deal with rollup? the selectedIndex is not part of key

Comment (morrySnow) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/StatementContext.java:158]: i think u just need to replace statistic derive result of mv olap scan by this value if it present?

Comment (seawinde) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/StatementContext.java:158]: selectedIndex  is a new scan, relationId is different from original scan, should be also getted from the `relationIdToStatisticsMap`

**Review by englefly — APPROVED** [review]

**Review by morrySnow — APPROVED** [review]

**Pull request merged**